### PR TITLE
Buffs Clogged Vents

### PR DESCRIPTION
**Pull request by Iamgoofball**
:cl: Iamgoofball
feature: Clogged Vents now has some teeth
/:cl:

This was the only part of GoofEvents players liked

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
4000u of a random chem.

Epic... simply epic.

**Comment by Iamgoofball** [comment]
hell yeah my dude

it's metal as fuck

clogged vents is dangerous now

**Comment by epochayur** [comment]
@Iamgoofball get a hobby

**Comment by JJRcop** [comment]
@epochayur I don't see anyone getting paid for this, especially not every day.

**Comment by epochayur** [comment]
@JJRcop my point being the he could be actually be doing something with his time as opposed to making 10 shit PRs in the span of 30 seconds

**Comment by Iamgoofball** [comment]
this isnt a shit PR though,this is one that Cheridan supported when it was PR'd initially

**Comment by GunHog** [comment]
I support this for the chaos it will cause. Does the proc you are using to select a reagent select *any* reagent in the game, or is it restricted in some fashion? (So you do not get vents making admin-foam or xeno transformation smoke)

File: code/modules/events/vent_clog.dm [hunk 1]
```diff
@@ -1,49 +1,39 @@
...
-			vent = pick_n_take(vents)
-
+/datum/round_event/vent_clog/start()
+	for(var/obj/machinery/atmospherics/components/unary/vent in vents)
 		if(vent && vent.loc)
-			var/datum/reagents/R = new/datum/reagents(50)
+			var/datum/reagents/R = new/datum/reagents(1000)
 			R.my_atom = vent
-			R.add_reagent(pick(gunk), 50)
+			R.add_reagent(pick(get_random_reagent_id()), 1000)
```
Comment (SpaceManiac) [inline code/modules/events/vent_clog.dm]: `add_reagent(get_random_reagent_id(), 1000)`, no need to `pick()` again

Comment (Iamgoofball) [reply code/modules/events/vent_clog.dm]: ty

**Comment by Iamgoofball** [comment]
@GunHog It prevents adminord and anything we set to can_synth 0.

**Comment by outbools** [comment]
>web editor

you fuckin monster

**Comment by Buggy123** [comment]
:+1: clogged vents is the tamest fucking event in the entire game, nice to see it get some # TEETH

**Comment by ma44** [comment]
inb4 slime mutation toxin, morph transformation reagent, nanomachines (oh hey robots), tuberculosis and syndicate nanites suddenly appear.

**Comment by Anonmare** [comment]
Bear in mind this can let it produce the various acids and acid can melt machines, like cameras/air alarms/APCs/etc.

**Comment by epochayur** [comment]
it's almost like filling the vents with random chems is a terrible idea

**Comment by Incomptinence** [comment]
quit repeating yourself weak banana it sounds good

**Comment by davethwave** [comment]
perhaps instead make the event a 2 tier system which this one is rarer and the non buffed one is more common.

**Comment by Iamgoofball** [comment]
No

**Comment by kevinz000** [comment]
dumb meme but good dumb meme :+1:

**Comment by Jalleo** [comment]
Well if it does smoke out the scrubbers at the SM and literally melt the SM nobody will be shocked

**Comment by Buggy123** [comment]
I don't think you can melt the SM.

Sorry lemme rephrase, I _really really hope_ you can't melt the SM.

**Comment by 81Denton** [comment]
4000u of random chems? Sign me up, this is gonna be fun,

**Comment by davethwave** [comment]
I'll purposefully go into the smoke to seek the hope of getting some nanomachines in me and die within a minute due to poison.

File: code/modules/events/vent_clog.dm [hunk 2]
```diff
@@ -1,49 +1,39 @@
...
-			vent = pick_n_take(vents)
-
+/datum/round_event/vent_clog/start()
+	for(var/obj/machinery/atmospherics/components/unary/vent in vents)
 		if(vent && vent.loc)
-			var/datum/reagents/R = new/datum/reagents(50)
+			var/datum/reagents/R = new/datum/reagents(1000)
 			R.my_atom = vent
-			R.add_reagent(pick(gunk), 50)
+			R.add_reagent(get_random_reagent_id(), 1000)
```
Comment (JJRcop) [inline code/modules/events/vent_clog.dm]: This doesn't look like 4000u to me

Comment (KieranCat) [reply code/modules/events/vent_clog.dm]: then fix it and make your own pr

Comment (81Denton) [reply code/modules/events/vent_clog.dm]: Smoke/foam carry 4x the reagents now if I'm correct

Comment (JJRcop) [reply code/modules/events/vent_clog.dm]: Oh!

Comment (81Denton) [reply code/modules/events/vent_clog.dm]: @JJRcop  #34485

**Comment by calzilla1** [comment]
>be warden
>clogged vents
>cl3f smoke in office and plasma in the brig
>ggnore
Merge immediately

File: code/modules/events/vent_clog.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/events/vent_clog.dm]: All is well!

When will github allow you to cancel a review though :thinking:

**Comment by ExcessiveUseOfCobblestone** [comment]
The funny thing is at this amount (especially with the recent chem buff that x4's it) even stypic and silversulf foam will kill you, and literally every healing chem except perf maybe will OD you if it doesn't have an addiction to the point of death.

Even when you win the chem roll you still lose lol.

There is no way this will end badly unless laughing causes IRL physical pain.

**Comment by GunHog** [comment]
I tested this, and proceeded to roam around a bit. I was filled with hot chocolate, flour, and nearby I got spammed with messages of books being erased by alcohol foam. I eventually entered green foam, which turned me into a slime person. This is 100% a Goof style chaos PR!

**Comment by ExcessiveUseOfCobblestone** [comment]
SM delam is FUN!

**Comment by Iamgoofball** [comment]
I didn't even know the supermatter had chemical reactions

**Comment by 81Denton** [comment]
That's-a spicy PR
Might wanna exclude SM/incinerator/general engine vents from that event

**Comment by ExcessiveUseOfCobblestone** [comment]
@Iamgoofball it doesn't iirc but if you break the floor tiles into space the SM gets very sad.

**Comment by GunHog** [comment]
I think the problem is with **cockroach spawns.** They spawn, then get eaten by the previously inert SM crystal, bringing it to life. Since I had not set up the engine, the crystal just keeps farting gases while the gas loop remains in roundstart state. This is potentially not an issue in most rounds where the SM is correctly activated and vented.

**Comment by Iamgoofball** [comment]
Ahahaha

We should make cockroaches immune to the SM to continue the joke re: nukes

**Comment by Bawhoppen** [comment]
This is maybe a little too crazy...

**Comment by Iamgoofball** [comment]
Nope

**Comment by vuonojenmustaturska** [comment]
I like this on principle but a random event delaming the SM and deleting departments via turf/machine deleting acids is a bit too crazy

**Comment by Buggy123** [comment]
Its fiiiiine, its fiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiine.

**Comment by Tlaltecuhtli** [comment]
wow it makes engineers rebuilding machines something that may happen???

**Comment by 81Denton** [comment]
Nah the cap will have called the shuttle before any engineer can get to tech storage

**Comment by GunHog** [comment]
>Nah the cap will have called the shuttle before any engineer can get to tech storage

You mean the room filled with ashes?

**Comment by PKPenguin321** [comment]
Perhaps put this event on a severity scale, like meteors? You see the clogged vents announcement and it's a gamble if it just poots water or spews lava

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]
Regarding delaming the SM, I believe you said you were fixing it so I'll not nag about it further.

Regarding ClF3, I will not merge an extremely common, "safe" event that deletes departments that cannot be rebuilt, especially as it has no occurrence limit and you are making it more common. Over 50% chance threatening/catastrophic meteos-level damage on the station is too much. 

However, I am not categorically against ClF3 from an event like this, maybe you could make it a subtype with proper occurrence limits and minimum player values and shit. Look at https://github.com/tgstation/tgstation/blob/b2b3f7f7c0cbb357a5452bc6f31e3f8ee2b7d9ab/code/modules/events/meteor_wave.dm#L55-L61 or the rod event for an example

**Comment by duncathan** [comment]
hell yeah

**Comment by Chocodaemon** [comment]
Just curious, how many times would this happen on a round, on average?

**Comment by duncathan** [comment]
@Chocodemon once, and only once.

**Comment by LetterJay** [comment]
Good job.
image

**Comment by Iamgoofball** [comment]
>CITADEL STATION
@LetterJay 
Looks like your PR mirroring bot forgot to mirror the nerf
tell them to merge the nerf lmao

@Poojawa your bot is broken

https://github.com/tgstation/tgstation/pull/35317

**Comment by Qbopper** [comment]
it's another "goof makes a pr that sounds interesting but the execution is dumb and it wasn't tested" episode

**Comment by CitrusGender** [comment]
dreamseeker_2018-02-13_19-02-52

This is really getting out of hand.
It's ended a few rounds just by happening and it has a tendency to light areas on fire.
Either turn down the frequency of the event or revert, this has been causing a lot of problems.

**Comment by Qbopper** [comment]
>it's another "goof makes a pr that sounds interesting but the execution is dumb and it wasn't tested" episode

also it ruins rounds

WHODATHUNK

**Comment by DaedalusGame** [comment]
make this a secrets panel button

**Comment by rustledjimm** [comment]
This needs to be nerfed or it needs to be tiered like asteroids are. Only on it's catastrophic setting will you get the chance of certain chemicals.

**Comment by epochayur** [comment]
WOW, IF ONLY SOMEONE HAD SAID THIS WOULD BE A TERRIBLE IDEA

**Pull request merged**